types/wlr_scene.c

```diff
@@ -226,6 +276,27 @@ static void scissor_output(struct wlr_output *output, pixman_box32_t *rect) {
 	wlr_renderer_scissor(renderer, &box);
 }
 
+static void render_rect(struct wlr_output *output,
+		pixman_region32_t *output_damage, const float color[static 4],
+		const struct wlr_box *box, const float matrix[static 9]) {
+	struct wlr_renderer *renderer = wlr_backend_get_renderer(output->backend);
+	assert(renderer);
+
+	pixman_region32_t damage;
+	pixman_region32_init(&damage);
+	pixman_region32_init_rect(&damage, box->x, box->y, box->width, box->height);
+	pixman_region32_intersect(&damage, &damage, output_damage);
+	if (pixman_region32_not_empty(&damage)) {
```

Comment:
Technically, no need for this condition. If the region is empty the `for` loop will be a no-op, and the `if` doesn't avoid intermediate computations.

Reply:
Does that mean it can be removed from the existing `render_texture()` as well?

Reply:
Very likely yes! Feel free to add a separate commit to do this, either in this PR or in a separate one.